### PR TITLE
First few tests of the Board module

### PR DESCRIPTION
**Pull request by sarahabimay**
@maikon
@jsuchy

### CODE REVIEW

File: ttt/lib/ttt/board.ex [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def get_move_count_per_mark(board, marks) do
+    Enum.map(marks, fn(mark) ->
+      %{mark => Enum.count(board, fn(move) -> move == mark end)}
+    end)
+  end
+
+  def mark_with_fewest_moves(move_counts_per_mark) do
+    List.to_string(Map.keys(Enum.min_by(move_counts_per_mark, fn(x) -> Map.values(x) end)))
```
Comment (Maikon) [inline ttt/lib/ttt/board.ex:22]: This function is doing too many things. Also prefer the `|>`when combining multiple functions like you did below.

File: ttt/lib/ttt/board.ex [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def game_over?(board, marks) do
+    remaining_spaces?(board, marks)
+  end
+
+  def remaining_spaces?(board, marks) do
+    sum_of_all_mark_moves(board, marks) < length board
```
Comment (Maikon) [inline ttt/lib/ttt/board.ex:30]: Keep consistency with whichever convention you're using, calling a function with `()` or without `()`.

File: ttt/test/board_test.exs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    board = [row1, row2, row3]
+    marks = ["X", "O"]
+    assert TTT.Board.next_mark_to_play(List.flatten(board), marks) == "O"
+  end
+
+  test "game is over if no more positions remaining" do
+    row1 = ["X", "O", "X"]
+    row2 = ["O", "X" ,"X"]
+    row3 = ["O", "X", "O"]
+    board = [row1, row2, row3]
```
Comment (Maikon) [inline ttt/test/board_test.exs:41]: Why is the board here two-dimensional vs line 12 above where it's flat? What would happen if I called `game_over?` with the flat array?

File: ttt/lib/ttt/board.ex [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  end
+
+  def sum_of_all_mark_moves(board, marks) do
```
Comment (Maikon) [inline ttt/lib/ttt/board.ex:33]: Does the board need to know about all possible marks? Same question applies to functions above that have the same signature.

File: ttt/lib/ttt/board.ex [hunk 5]
```diff
@@ -0,0 +1,41 @@
+defmodule TTT.Board do
+  def empty_board(dimension) do
+   Enum.to_list(1..dimension*dimension)
+  end
+
+  def play_move(board, move, mark) do
+    List.replace_at(board, move - 1, mark)
+  end
+
+  def next_mark_to_play(board, marks) do
```
Comment (Maikon) [inline ttt/lib/ttt/board.ex:10]: Could we figure out what the next mark to play is by only using the board?

File: ttt/lib/ttt/board.ex (hunk 4 above)
Comment (sarahabimay) [inline ttt/lib/ttt/board.ex:33]: Yes the marks list is currently ["X", "O"] but could potentially be ["SARAH", "MAKIS"] so I need to pass that info each time

File: ttt/lib/ttt/board.ex (hunk 5 above)
Comment (sarahabimay) [inline ttt/lib/ttt/board.ex:10]: Yes I'm sure with a bunch of Enum or List functions.. but I'm not sure how yet.

File: ttt/test/board_test.exs (hunk 3 above)
Comment (sarahabimay) [inline ttt/test/board_test.exs:41]: I'm in two minds yet whether to make it 2D or 1D.. hence the mix..

File: ttt/lib/ttt/board.ex (hunk 1 above)
Comment (sarahabimay) [inline ttt/lib/ttt/board.ex:22]: I spotted that and changed it

**Pull request merged**